partitura/utils/music.py

```diff
@@ -3330,6 +3330,195 @@ def step2pc(step, alter):
+                temp_ppart = list_pparts.pop()
+                first_note_temp_ppart = temp_ppart.notes[0]
+                start_time = first_note_temp_ppart['note_on'] # take the previous segment's start time while keeping the present segment's end time
+                ppart_segment = slice_ppart_by_time(ppart, start_time, end_time)
+        
+        else:
+            segment_jump_back_counter = 0
+            ppart_segment = slice_ppart_by_time(ppart, start_time, end_time)
+
+        # from the second segment onwards, add the last control values from the previous segment to the current segment.
+        if len(list_pparts) != 0: 
+            segment_first_note = ppart_segment.notes[0]
+            for cc in last_control_values:
+                first_cc = {}
+                first_cc['number'] = cc
+                first_cc['value'] = last_control_values[cc]['value']
+                first_cc['track'] = last_control_values[cc]['track']
+                first_cc['channel'] = last_control_values[cc]['channel']
+                first_cc['time'] = segment_first_note['note_on']
+                first_cc['time_tick'] = segment_first_note['note_on_tick']
+                ppart_segment.controls.insert(0, first_cc)
+        
+        last_control_values = get_last_cc_values(ppart_segment.controls, cc_nums_in_ppart)
+        list_pparts.append(ppart_segment)
+
+        start_end_times.append((start_time, end_time))
+
+    return list_pparts, start_end_times
+
+
+def segment_ppart_by_gap(
+        ppart: PerformedPart,
+        gap: float = 8.0,
+        min_segment_duration: float = 10.0
+        ) -> Tuple[list, list]:
+    """
+    Segment a performed part by a given gap (in seconds) between the 'note_on' values of 
+    consecutive notes and return a list of the segmented performed parts.
+
+    Parameters
+    ----------
+    ppart : PerformedPart
+        A PerformedPart object representing the performed part.
+    gap : float
+        The minimum gap between notes to split the segment. The default is 8.0.
+    min_segment_duration : float
+        The minimum duration of a segment in seconds. The default is 10.0.
+
+    Returns
+    -------
+    Tuple[list_pparts, start_end_times]:
+    
+        list_pparts : list
+            A list of PerformedPart objects representing the segmented performed parts.
+        start_end_times : list
+            A list of tuples where each tuple contains the start and end times of each
+            segment in the performed part. This is useful to later split the corresponding audio files.
+    """
+
+    perf_notes = ppart.notes
+
+    # find the indices of the notes in perf_notes where the gap between 
+    # consecutive notes' 'note_on' and 'note_off' values is greater than the threshold.
+    note_ons = np.array([note['note_on'] for note in perf_notes])
+    sound_offs = np.array([note['sound_off'] for note in perf_notes]) # TODO: Consider changing to 'sound_off' instead of 'note_off'
```

Comment:
I think it would be worthwhile (and more intuitive for users) to use "note_off" instead of "sound_off". Maybe we can have a flag for this?